library/network_connections.py

```diff
@@ -336,6 +336,7 @@ def ifcfg_create(
             ifcfg["ONBOOT"] = "no"
 
         ifcfg["DEVICE"] = connection["interface_name"]
+        ifcfg["MACADDR"] = connection["cloned_mac"]
```

Comment:
setting this to special values like `preserve` wouldn't work... the role should do something about this.

for example, if the initscripts provider is used, then `validate_connection_one()` would handle that (e.g. by logging a warning and ignoring the unsupported value, or failing the playbook as invalid altogether).

Reply:
added an exception in `validate_connection_one()`